staging/src/k8s.io/apimachinery/pkg/util/net/port_range.go

```diff
@@ -47,31 +47,64 @@ func (pr PortRange) String() string {
 // sets the PortRange from it.  This is part of the flag.Value and pflag.Value
 // interfaces.
 func (pr *PortRange) Set(value string) error {
-	value = strings.TrimSpace(value)
+	const (
+		SinglePortNotation = iota
+		HyphenNotation     = 1 << iota
+		PlusNotation
+	)
```

Comment:
nit, can you define this instead:
``` golang
	const (
		SinglePortNotation = 1 << iota
		HyphenNotation
		PlusNotation
	)
```

The current definition doesn't cause any real damage, but it tortures me to think the array is "0, 2, 4, 8, 16..."

Reply:
Yeah it does torture (facepalm

Reply:
Its a little tricky though. Defining SinglePortNotation to `0` makes `SinglePortNotation` a default case. Also applying `or` operand to the `notation` can help detect the case when `hyphenIndex` and `plusIndex` are not present at the same time. This makes the code a lot cleaner but maybe hard to read.

Reply:
Sorry, I don't get your point. `SinglePortNotation = 1 << iota` makes `SinglePortNotation` 1, not 0.

I don't think you need to change the `|=`. What did I miss?

Reply:
@caesarxuchao I was talking about the previous code.